modules/redis/scanner.go

```diff
@@ -152,6 +169,71 @@ func (scan *scan) Close() {
 	defer scan.close()
 }
 
+func getUnmarshaler(file string) (func([]byte, interface{}) error, error) {
+	var unmarshaler func([]byte, interface{}) error
+	switch ext := filepath.Ext(file); ext {
+	case ".json":
+		unmarshaler = json.Unmarshal
+	case ".yaml", ".yml":
+		unmarshaler = yaml.Unmarshal
+	default:
+		err := fmt.Errorf("File type %s not valid.", ext)
+		return nil, err
+	}
+	return unmarshaler, nil
+}
+
+func getFileContents(file string, output interface{}) error {
+	unmarshaler, err := getUnmarshaler(file)
+	if err != nil {
+		return err
+	}
+	fileContent, err := ioutil.ReadFile(file)
+	if err != nil {
+		return err
+	}
+	err = unmarshaler([]byte(fileContent), output)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Initializes the command mappings
+func (scanner *Scanner) initCommands() error {
+	scanner.commandMappings = map[string]interface{}{
+		"PING":        "PING",
+		"AUTH":        "AUTH",
+		"INFO":        "INFO",
+		"NONEXISTENT": "NONEXISTENT",
+		"QUIT":        "QUIT",
+	}
+
+	if scanner.config.CustomCommands != "" {
+		var customCommands []string
+		err := getFileContents(scanner.config.CustomCommands, &customCommands)
+		if err != nil {
+			return err
+		}
+		scanner.customCommands = customCommands
+	}
+
+	// User supplied a file for updated command mappings
+	if scanner.config.Mappings != "" {
+		var mappings map[string]string
+		err := getFileContents(scanner.config.Mappings, &mappings)
+		if err != nil {
+			return err
+		}
+		for origCommand, newCommand := range mappings {
+			scanner.commandMappings[strings.ToUpper(origCommand)] = strings.ToUpper(newCommand)
```

Comment:
Are commands required to be uppercase? If so, that should be called out; if not, maybe mention in that the help.

Reply:
No they're not. Redis commands are case insensitive, so sending `info` is the same as sending `INFO`. Since the commands we send are all uppercase, if the mapping updates a user passes in is `"info":"renamed_info"`, I'd want the command that we send when scanning for 'INFO' to be the new mapping. I basically did this to ensure that our commands get overwritten correctly in the case where the the inputted JSON/YAML file has lowercase commands

Reply:
got it. This makes perfect sense then.